File: javascript/easy/distributeCandies.js
/**
 * @param {number[]} candies
 * @return {number}
 */
var distributeCandies = function(candies) {
    const set = new Set(), share = parseInt(candies.length/2);
    for(let candy of candies) {
        set.add(candy);
    }
    return (set.size > share) ? share : set.size;
};

File: ruby/easy/distribute_candies.rb
# @param {Integer[]} candies
# @return {Integer}
def distribute_candies(candies)
    set = Set.new
    candies.each do |candy|
        set.add(candy)
    end
    return (set.length > (candies.length/2)) ? (candies.length/2) : set.length
end
